sympy/core/expr.py

```diff
@@ -1619,22 +1619,31 @@ def find(l, sub, first=True):
                         return Add(*[Mul(*(list(r) + n[:-len(end) + ii])) for r, n in co])
                     else:
                         return Mul(*end[ii + len(nx):])
-            # look for single match
-            hit = None
+            # look for a single hit but allow a sum of terms if each have the
+            # same remaining expression after removing coefficient
+            hit = []
             for i, (r, n) in enumerate(co):
                 ii = find(n, nx, right)
                 if ii is not None:
                     if not hit:
-                        hit = ii, r, n
+                        if not right:
+                            np = n[ii+len(nx):]
+                        else:
+                            np = n[:ii]
                     else:
-                        break
-            else:
-                if hit:
-                    ii, r, n = hit
+                        if not right:
```

Comment:
Minor thing, but I'd swap the order of these if-else clauses so that the negation is avoided.

Reply:
I did it this way because whoever wrote the function originally wrote if-else clauses with negation(for example line 1594 and a few other places.) I fixed my own contribution to avoid the negation. Should I also change the other occurrences of this in the coeff() function which I did not write?

Reply:
I think it is enough to change it here (although not required). Unless there is a "pedagogical" benefit of starting with the not-clause it is marginally better not to. But there is indeed a lot of code where it is not like that. I actually tried the other day to fix it systematically, but in some parts of the code base, it really needs to be written with not first...